cc-pk-service/src/handleData.ts

```diff
@@ -0,0 +1,200 @@
+*/
+export async function getSurveys() {
+    try {
+        const response = axios.get<Survey[]>('https://api.purkukartoitus.fi/v1/surveys', {
+            params: {
+                'maxResults': '2147483647'
+            },
+            headers: {
+                'Authorization': 'Bearer ' + userToken
+            }
+        });
+
+        // Find which surveys' belong to the user
+        const allSurveys:Survey[] = (await response).data;
+        const surveys:Survey[] = [];
+        for (let i = 0; i < allSurveys.length; i++) {
+            if (allSurveys[i].metadata.creatorId === await (getId()) && allSurveys[i].status === 'DONE') surveys.push(allSurveys[i]);
+        }
+        return surveys;
+    } catch (err) {
+        console.log('Error: cannot fetch surveys');
+        return [];
+    }
+}
+
+/**
+ * Collects user's reusable items from each survey
+ * @param token User's token for accessing PK API
+ * @param userSurveys User's surveys
+ * @returns Array of Item's that belong to the user
+ */
+export async function getReusables(userSurveys:Survey[]) {
+    try {
+        const items:Item[] = [];
+        for (let i = 0; i < userSurveys.length; i++) {
+            // Collect PK API's items that belong to the survey
+            const response = axios.get<ItemInfo[]>(`https://api.purkukartoitus.fi/v1/surveys/${userSurveys[i].id}/reusables`, 
+            {
+                headers: {
+                    'Authorization': 'Bearer ' + userToken
+                }
+            });
+            const fetchedItems:ItemInfo[] = (await response).data;
+  
+            // Collect PK API's buildings that belong to the survey
+            const responseBuilding = axios.get<Building[]>(`https://api.purkukartoitus.fi/v1/surveys/${userSurveys[i].id}/buildings`, 
+            {
+                headers: {
+                    'Authorization': 'Bearer ' + userToken
+                }
+            });
+            const fetchedBuildings:Building[] = (await responseBuilding).data;
+  
+            // Create a new item from the fetched information
+            for (let j = 0; j < fetchedItems.length; j++) {
+                const item:Item = {
+                reusableId: fetchedItems[j].id,
+                componentName: fetchedItems[j].componentName,
+                surveyId: userSurveys[i].id,
+                streetAddress: fetchedBuildings[0].address.streetAddress
+            }
+            items.push(item);
+            }
+        }
+        return items;
+    } catch (err) {
+      console.log('Error: cannot fetch reusable items');
+      return [];
+    }
+}
+
+/**
+ * Collects information about user's item
+ * @param token User's token for accessing PK API
+ * @param userSurveys User's surveys
+ * @param itemId Id of the Item that we want information about
+ * @returns ItemInfo -object that includes all the info PK API has to offer about the item
+ */
+export async function getItemInfo(userSurveys:Survey[], itemId:string) {

```

Comment:
This function always returns initInfo. The loops dont save/collect data anywhere?